tests/test_grid.py

```diff
@@ -920,6 +920,59 @@ def test_cast_uniform_to_rectilinear():
     assert rectilinear.bounds == grid.bounds
 
 
+def test_fft_and_rfft(noise_2d):
+    grid = pyvista.UniformGrid(dims=(10, 10, 1))
+    with pytest.raises(ValueError, match='active point scalars'):
+        grid.fft()
+
+    name = noise_2d.active_scalars_name
+    noise_fft = noise_2d.fft()
+    assert noise_fft[name].dtype == np.complex128
+
+    full_pass = noise_2d.fft().rfft()
+    assert full_pass[name].dtype == np.complex128
+
+    # expect FFT and and RFFT to transform from time --> freq --> time domain
+    assert np.allclose(noise_2d['scalars'], full_pass[name].real)
+    assert np.allclose(full_pass[name].imag, 0)
+
+    output_scalars_name = 'out_scalars'
+    noise_fft = noise_2d.fft(output_scalars_name=output_scalars_name)
+    assert output_scalars_name in noise_fft.point_data
+
+
+def test_fft_low_pass(noise_2d):
+    name = noise_2d.active_scalars_name
+    noise_no_scalars = noise_2d.copy()
+    noise_no_scalars.clear_data()
+    with pytest.raises(ValueError, match='FFT filters require'):
+        noise_no_scalars.low_pass(1, 1, 1)
+
+    noise_too_many_scalars = noise_no_scalars.copy()
+    noise_too_many_scalars.point_data.set_array(np.arange(noise_2d.n_points), 'a')
+    noise_too_many_scalars.point_data.set_array(np.arange(noise_2d.n_points), 'b')
+    with pytest.raises(ValueError, match='There are multiple point scalars available'):
+        noise_too_many_scalars.low_pass(1, 1, 1)
+
+    with pytest.raises(ValueError, match='must be complex data'):
+        noise_2d.low_pass(1, 1, 1)
+
+    out_zeros = noise_2d.fft().low_pass(0, 0, 0)
+    assert np.allclose(out_zeros[name][1:], 0)
+
+    out = noise_2d.fft().low_pass(1, 1, 1)
+    assert not np.allclose(out[name][1:], 0)
```

Comment:
I still don't completely get the `out[name][1:]` here. Is the idea that the first scalar belongs to the `(0, 0, 0)` frequency, and all the other points should then be zero after this filter? Because if so, I don't understand why we're testing the same `out_zeros[name][1:]` in the high-pass case.

Or is there some other reason why the first scalar is ignored in these tests?

Reply:
Only reason is a copy/pasta from the low_pass. Fixed in 45b20759a970a65f1e6d503156f03b9ddf5a91f2